### PR TITLE
Feature/wf 443-colors

### PR DESCRIPTION
**Pull request by geroldboehler**

### CODE REVIEW

File: app/styleguide/components/s-color-list.vue [hunk 1]
```diff
@@ -0,0 +1,66 @@
+<template>
+  <div :class="b()">
+    <div v-for="palette in palettes" :key=palette>
+      <h4 :class="b('palette-headline')">{{palette}} colors</h4>
+      <div :class="b('grid')">
+        <div v-if="palette !== 'gradient'" v-for="color in colors[palette]" :class="b('grid-item')" :key=color>
```
[the remark this answers is not in the document]
Comment (geroldboehler) [reply app/styleguide/components/s-color-list.vue]: Sorry, completely forgot to check lint warnings.

File: app/styleguide/components/s-color-item.vue [hunk 2]
```diff
@@ -0,0 +1,107 @@
+<template>
+  <div :class="b('grid-item')">
+    <div :class="b('color-wrapper')">
+      <div :class="b('color')" :style="backgroundColor"></div>
```
Comment (patric-eberle) [inline app/styleguide/components/s-color-item.vue]: I would recommend object syntax when using v-bind so it is easier to extend: https://vuejs.org/v2/guide/class-and-style.html#Object-Syntax-1

File: app/styleguide/components/s-color-item.vue [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+    }
+
+    &__color {
+      width: 240px;
+      height: 180px;
+    }
+
+    &__name {
+      text-align: center;
+      font-size: 14px;
```
Comment (patric-eberle) [inline app/styleguide/components/s-color-item.vue]: - [ ] Use font mixin and variables (you can also define styleguide variables, if you think it makes sense)

File: app/styleguide/components/s-color-item.vue [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+      line-height: 24px;
+      font-weight: bold;
+    }
+
+    &__value {
+      text-align: center;
+      font-size: 12px;
```
Comment (patric-eberle) [inline app/styleguide/components/s-color-item.vue]: - [ ] Use font mixin and variables (you can also define styleguide variables, if you think it makes sense)

File: app/styleguide/components/s-palette-item.vue [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+    // destroyed() {},
+
+    // methods: {},
+    // render() {},
+  };
+</script>
+
+<style lang="scss">
+  .s-palette-item {
+    display: inline-block;
```
Comment (patric-eberle) [inline app/styleguide/components/s-palette-item.vue]: - [ ] Why `inline-block` when you define a width of `100%` anyway? Recommend to use default (`block`) and no width.

File: app/styleguide/components/s-palette-list.vue [hunk 6]
```diff
@@ -0,0 +1,66 @@
+<template>
+  <div :class="b()">
+    <s-palette-item
```
Comment (patric-eberle) [inline app/styleguide/components/s-palette-list.vue]: - [ ] Please always use a wrapper when placing a component inside an other.

```
<div :class="b('item')">
  <s-palette-item ... />
</div>
```

This way spacing and orientation can be defined.

File: app/styleguide/components/s-palette-item.vue [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+    width: 100%;
+
+    &__grid {
+      display: flex;
+      flex-wrap: wrap;
+      margin: 0 -5px;
```
Comment (patric-eberle) [inline app/styleguide/components/s-palette-item.vue]: - [ ] Use spacing variables.

File: app/styleguide/components/s-palette-list.vue [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+      v-for="palette in palettes"
+      :palette="palette"
+      :colors="colors[palette]"
+      :key="palette"/>
+  </div>
+</template>
+
+<script>
+
+  import _colors from '../../setup/js/_color';
```
Comment (patric-eberle) [inline app/styleguide/components/s-palette-list.vue]: - [ ] Please don't use underline in JS. In SCSS it is common convention to mark include only files with an underline at the beginning. Don't apply this to JS.

File: app/styleguide/components/s-palette-list.vue [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+  export default {
+
+    name: 's-palette-list',
+    components: {
+      's-palette-item': sPaletteItem
+    },
+    // mixins: [],
+
+    props: {
+      palette: {
```
Comment (patric-eberle) [inline app/styleguide/components/s-palette-list.vue]: - [ ] Add property documentation

File: app/styleguide/components/s-palette-list.vue [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+
+    // methods: {},
+    // render() {},
+  };
+
+</script>
+
+<style lang="scss">
+  .s-palette-list {
+    font-family: "Helvetica Neue", Helvetica, Arial, sans-serif;
```
Comment (patric-eberle) [inline app/styleguide/components/s-palette-list.vue]: - [ ] Yes, please define styleguide variables ;)

File: app/styleguide/core/grid/grid.md [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+| xl   | 1440px     |
+
+### Media queries
+Use the media-mixin for media queries. See app/setup/scss/mixins/_media.scss
+
+```scss
+@mixin media($up: null, $down: null, $media: all, $breakpoints: $grid-breakpoints)
+```
+
+Use media-mixin within styles, not the other way around
```
Comment (patric-eberle) [inline app/styleguide/core/grid/grid.md]: - [ ] Use `selectors` instead of `styles`

File: app/styleguide/routes/index.vue [hunk 12]
```diff
@@ -34,17 +34,17 @@
 <style lang="scss">
   .index {
     &__warning {
-      border: 1px solid $color-status--failed;
-      background: rgba($color-status--failed, 0.5);
+      border: 1px solid $color-status--danger;
+      background: rgba($color-status--danger, 0.5);
       text-align: center;
       padding: 10px;
```
Comment (patric-eberle) [inline app/styleguide/routes/index.vue]: - [ ] Use `$spacing` variable

File: app/styleguide/components/s-color-item.vue [hunk 13]
```diff
@@ -0,0 +1,107 @@
+<template>
```
[the remark this answers is not in the document]
Comment (geroldboehler) [reply app/styleguide/components/s-color-item.vue]: Thank you, next time i take more time until i submit it for review ;)

File: app/styleguide/components/s-color-item.vue [hunk 14]
```diff
@@ -0,0 +1,107 @@
...
+        return this.color.name.split(' - ').join('<br>');
+      },
+      value() {
+        return this.color.value.join(' - ');
+      },
+      backgroundColor() {
+        let [color1, color2] = this.color.value;
+
+        if (!color2) {
+          color2 = color1;
```
[the remark this answers is not in the document]
Comment (geroldboehler) [reply app/styleguide/components/s-color-item.vue]: I knew there was a better solution :)

File: app/styleguide/components/s-palette-list.vue (hunk 10 above)
Comment (geroldboehler) [reply app/styleguide/components/s-palette-list.vue]: I will fix that in the WF-441_fonts branch

File: app/styleguide/components/s-color-item.vue [hunk 15]
```diff
@@ -0,0 +1,107 @@
...
+</template>
+
+<script>
+  export default {
+    name: 's-color-item',
+    // components: {},
+    // mixins: [],
+
+    props: {
+      color: {
```
[the remark this answers is not in the document]
Comment (geroldboehler) [reply app/styleguide/components/s-color-item.vue]: I couldn't figure it out, i changed the component to just receive the 2 attributes.

File: app/styleguide/components/s-color-item.vue (hunk 2 above)
Comment (geroldboehler) [reply app/styleguide/components/s-color-item.vue]: Fixed

File: app/styleguide/components/s-color-item.vue (hunk 3 above)
Comment (geroldboehler) [reply app/styleguide/components/s-color-item.vue]: I will fix that in the WF-441_fonts branch

File: app/styleguide/components/s-color-item.vue (hunk 4 above)
Comment (geroldboehler) [reply app/styleguide/components/s-color-item.vue]: I will fix that in the WF-441_fonts branch

File: app/styleguide/components/s-palette-list.vue (hunk 6 above)
Comment (geroldboehler) [reply app/styleguide/components/s-palette-list.vue]: Not sure if i understand, lets discuss it on tuesday.

**Comment by geroldboehler** [comment]
Hi Patric, i have fixed most of the issues. For now, only responsive issues seem to be open which i would like to discuss with you on tuesday.

File: app/styleguide/components/s-color-item.vue [hunk 16]
```diff
@@ -0,0 +1,119 @@
...
+        default: null,
+        required: true
+      }
+    },
+    // data() {
+    //   return {};
+    // },
+
+    computed: {
+      display_name() {
```
Comment (patric-eberle) [inline app/styleguide/components/s-color-item.vue]: - [ ] Please `always` use camelCase in JS.

File: app/styleguide/components/s-color-item.vue [hunk 17]
```diff
@@ -0,0 +1,119 @@
...
+        return this.name.split(' - ').join('<br>');
+      },
+      display_value() {
```
Comment (patric-eberle) [inline app/styleguide/components/s-color-item.vue]: - [ ] Please `always` use camelCase in JS.

File: app/styleguide/components/s-palette-list.vue [hunk 18]
```diff
@@ -0,0 +1,60 @@
...
+      v-for="palette in palettes"
+      :palette="palette"
+      :colors="colors[palette]"
+      :key="palette"/>
+  </div>
+</template>
+
+<script>
+
+  import _colors from '../../setup/js/color';
```
Comment (patric-eberle) [inline app/styleguide/components/s-palette-list.vue]: - [ ] Please don't use `underline` for variable names. In JS this often is seen as a private variable, which we don't use.

File: app/styleguide/core/grid/grid.md [hunk 19]
```diff
@@ -0,0 +1,32 @@
+### Available breakpoints
```
Comment (patric-eberle) [inline app/styleguide/core/grid/grid.md]: - [ ] Did you merge the current `develop` branch? I think these changes are from the merged variable branch and therefore should not show up here.

**Comment by geroldboehler** [comment]
Done. Also merged develop branch, please review again @patric-eberle

**Pull request merged**